src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/NetCoreApp3CompatModeTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.IO;
+using System.Linq;
+using BundleTests.Helpers;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Microsoft.DotNet.CoreSetup.Test;
+using Microsoft.NET.HostModel.Bundle;
+using Xunit;
+
+namespace AppHost.Bundle.Tests
+{
+    public class NetCoreApp3CompatModeTests : IClassFixture<NetCoreApp3CompatModeTests.SharedTestState>
+    {
+        private SharedTestState sharedTestState;
+
+        public NetCoreApp3CompatModeTests(SharedTestState fixture)
+        {
+            sharedTestState = fixture;
+        }
+
+        [Fact]
+        public void Bundle_Is_Extracted()
+        {
+            var fixture = sharedTestState.TestFixture.Copy();
+            string singleFile;
+            Bundler bundler = BundleHelper.BundleApp(fixture, out singleFile, BundleOptions.BundleAllContent);
+            var extractionBaseDir = BundleHelper.GetExtractionRootDir(fixture);
+
+            Command.Create(singleFile)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .EnvironmentVariable(BundleHelper.DotnetBundleExtractBaseEnvVariable, extractionBaseDir.FullName)
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World");
```

Comment:
Any way we could also confirm that the assemblies are actually loaded from that location? Perhaps by printing the result of `Assembly.Location` or similar?

Reply:
We should definitely test for this.

I think we always set `BUNDLE_PROBE` (regardless of 3.x compat mode), so bundle probing takes precedence during assembly loading - I could see some undesired behaviour happening as a result.

Reply:
With the most recent changes bundle probe should return false for all queries - I checked that the runtime behavior in that case is the same as without bundle probe in the case of assembly resolution. Note that it's not the same in other cases - especially Environment.GetCommandLineArguments[0] should still return the .exe path, not the main .dll path.

I also checked all the other places in runtime where we somehow react to bundle probe, and they all look correct to me.